stdlib/pervasives.gr

```diff
@@ -1,19 +1,46 @@
+import foreign wasm numberDecr : Number -> Number as decr from 'stdlib-external/runtime'
+import foreign wasm numberPlus : (Number, Number) -> Number as (+) from 'stdlib-external/runtime'
+import foreign wasm numberMinus : (Number, Number) -> Number as (-) from 'stdlib-external/runtime'
+import foreign wasm numberTimes : (Number, Number) -> Number as (*) from 'stdlib-external/runtime'
+import foreign wasm numberDivide : (Number, Number) -> Number as (/) from 'stdlib-external/runtime'
+import foreign wasm numberMod : (Number, Number) -> Number as (%) from 'stdlib-external/runtime'
 
 # Number comparisons
-primitive (<) : (Number, Number) -> Bool = "@less"
-primitive (>) : (Number, Number) -> Bool = "@greater"
-primitive (<=) : (Number, Number) -> Bool = "@lesseq"
-primitive (>=) : (Number, Number) -> Bool = "@greatereq"
+import foreign wasm numberLess : (Number, Number) -> Bool as (<) from 'stdlib-external/runtime'
+import foreign wasm numberGreater : (Number, Number) -> Bool as (>) from 'stdlib-external/runtime'
+import foreign wasm numberLessEqual : (Number, Number) -> Bool as (<=) from 'stdlib-external/runtime'
+import foreign wasm numberGreaterEqual : (Number, Number) -> Bool as (>=) from 'stdlib-external/runtime'
+
+# Number bit/logical operations
+import foreign wasm numberLnot : Number -> Number as lnot from 'stdlib-external/runtime'
+
+import foreign wasm numberLand : (Number, Number) -> Number as land from 'stdlib-external/runtime'
+import foreign wasm numberLor : (Number, Number) -> Number as lor from 'stdlib-external/runtime'
+import foreign wasm numberLxor : (Number, Number) -> Number as lxor from 'stdlib-external/runtime'
+
+import foreign wasm numberLsl : (Number, Number) -> Number as lsl from 'stdlib-external/runtime'
+import foreign wasm numberLsr : (Number, Number) -> Number as lsr from 'stdlib-external/runtime'
+import foreign wasm numberAsr : (Number, Number) -> Number as asr from 'stdlib-external/runtime'
+
+
+# Number coercions & conversions
+import foreign wasm coerceInt64ToNumber : Int64 -> Number from 'stdlib-external/runtime'
+import foreign wasm coerceInt32ToNumber : Int32 -> Number from 'stdlib-external/runtime'
+import foreign wasm coerceFloat64ToNumber : Float64 -> Number from 'stdlib-external/runtime'
+import foreign wasm coerceFloat32ToNumber : Float32 -> Number from 'stdlib-external/runtime'
+import foreign wasm coerceRationalToNumber : Rational -> Number from 'stdlib-external/runtime'
+import foreign wasm coerceNumberToInt64 : Number -> Int64 from 'stdlib-external/runtime'
+import foreign wasm coerceNumberToInt32 : Number -> Int32 from 'stdlib-external/runtime'
+import foreign wasm coerceNumberToRational : Number -> Rational from 'stdlib-external/runtime'
+import foreign wasm coerceNumberToFloat64 : Number -> Float64 from 'stdlib-external/runtime'
+import foreign wasm coerceNumberToFloat32 : Number -> Float32 from 'stdlib-external/runtime'
```

Comment:
Some of these are actually used by the desugaring process, and since literals can be made without importing a library, I think it makes sense to leave these here (in addition to creating the new & more easily memorable names you describe).